server.js

```diff
@@ -2,23 +2,84 @@ import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
 
+
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true, 
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date()
+  }
+})
+
+// createdAt - the time the Thought was added to the database
+// Defaults to the current time
+// Should not be assignable when creating a new thought
+
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
+const listEndpoints = require("express-list-endpoints");
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec();
+  res.json(thoughts);
+});
+
+app.post("/thoughts", async (req, res) => {
+  const {message} = req.body;
+  const thought = new Thought({message});
+
+  try {
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought);
+  } catch (err) {
+    res.status(400).json({message:'Could not save thought to the database', error:err.errors})
+  }
+});
+
+// New route to handle increasing hearts for a specific thought
+app.post("/thoughts/:thoughtId/like", async (req, res) => {
+  const { thoughtId } = req.params;
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(
+      thoughtId,
+      { $inc: { hearts: 1 } }, // Increment the hearts field by 1
+      { new: true } // Return the updated document
```

Comment:
It's interesting to see new ways to do the same thing that I also did! Here I used
`updateLike.hearts += 1; // Increment the 'hearts' property of the thought
    await updateLike.save(); // Save the updated thought`
Both works great but looks very different from eachother! Fun thing about code 😄